torchtext/experimental/datasets/raw/language_modeling.py

```diff
@@ -0,0 +1,146 @@
+import torch
+import logging
+import io
+from torchtext.utils import download_from_url, extract_archive
+
+URLS = {
+    'WikiText2':
+        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-2-v1.zip',
+    'WikiText103':
+        'https://s3.amazonaws.com/research.metamind.io/wikitext/wikitext-103-v1.zip',
+    'PennTreebank':
+        ['https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.train.txt',
+         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.test.txt',
+         'https://raw.githubusercontent.com/wojzaremba/lstm/master/data/ptb.valid.txt']
+}
+
+
+class RawTextIterableDataset(torch.utils.data.IterableDataset):
+    """Defines an abstraction for raw text iterable datasets.
+    """
+
+    def __init__(self, iterator, start=0, num_lines=None):
+        """Initiate language modeling dataset.
+        """
+        super(RawTextIterableDataset, self).__init__()
+        self._iterator = iterator
+        self.start = start
+        self.num_lines = num_lines
+
+    def __iter__(self):
+        for i, item in enumerate(self._iterator):
+            if i >= self.start:
+                yield item
+            if (self.num_lines is not None) and (i == (self.start + self.num_lines)):
+                break
+
+    def get_iterator(self):
+        return self._iterator
+
+
+def _get_datafile_path(key, extracted_files):
```

Comment:
Since `_get_datafile_path` is called only once, it may not be necessary to separate it from `_setup_datasets` func.